### PR TITLE
add isurance_entity

### PR DESCRIPTION
**Pull request by GhitSam**
Updated db_models.py with insurance class
Created insurance_crud.py
Updated response_models.py with insurance class baseModel
Update main.py with app.get for insurance

Screen Shot 2024-03-17 at 2 46 16 PM

### CODE REVIEW

**Comment by fbatroni** [comment]
Looks good. Couple of comments:
- You should always assign at least one person to review the code.
- In SQL, the default starting number for the SERIAL data type is 1. I'm curious why your insurance id and your patient_id is 0.

**Comment by AtVict0r** [comment]
> Looks good. Couple of comments:
> 
> * You should always assign at least one person to review the code.
> * In SQL, the default starting number for the SERIAL data type is 1. I'm curious why your insurance id and your patient_id is 0.

Thank you for your comments.

Regarding the code review, I apologize for the oversight. For some reason, the option to add a reviewer didn't show up when we created the pull request. We will ensure to assign at least one person for code review in the future.

As for the `insurance_id` and `patient_id` starting at 0, it's due to the dependencies in our entity creation. The entity we were assigned depends on the creation of the patient entity, and we encountered issues seeding our table because of this. Therefore, the screenshot attached is of the API docs and not an actual API call. Our API call resulted in an empty array, because we have not yet seeded the table.

File: lib/response_models.py
Comment (AtVict0r) [inline lib/response_models.py:1]: @fbatroni What does the `class Config` on line 29 and 47 do?
Do we need to add it below line 54?

Comment (fbatroni) [reply lib/response_models.py:1]: Yes - Class config is if you wanted to configure the model with some additional features. 
In our case the orm setting is for integration with object-relational mappings (ORMs) - the SQLAlchemy models.
https://docs.pydantic.dev/1.10/usage/model_config/

**Comment by fbatroni** [comment]
> > Looks good. Couple of comments:
> > 
> > * You should always assign at least one person to review the code.
> > * In SQL, the default starting number for the SERIAL data type is 1. I'm curious why your insurance id and your patient_id is 0.
> 
> Thank you for your comments.
> 
> Regarding the code review, I apologize for the oversight. For some reason, the option to add a reviewer didn't show up when we created the pull request. We will ensure to assign at least one person for code review in the future.
> 
> As for the `insurance_id` and `patient_id` starting at 0, it's due to the dependencies in our entity creation. The entity we were assigned depends on the creation of the patient entity, and we encountered issues seeding our table because of this. Therefore, the screenshot attached is of the API docs and not an actual API call. Our API call resulted in an empty array, because we have not yet seeded the table.



> > Looks good. Couple of comments:
> > 
> > * You should always assign at least one person to review the code.
> > * In SQL, the default starting number for the SERIAL data type is 1. I'm curious why your insurance id and your patient_id is 0.
> 
> Thank you for your comments.
> 
> Regarding the code review, I apologize for the oversight. For some reason, the option to add a reviewer didn't show up when we created the pull request. We will ensure to assign at least one person for code review in the future.
> 
> As for the `insurance_id` and `patient_id` starting at 0, it's due to the dependencies in our entity creation. The entity we were assigned depends on the creation of the patient entity, and we encountered issues seeding our table because of this. Therefore, the screenshot attached is of the API docs and not an actual API call. Our API call resulted in an empty array, because we have not yet seeded the table.

Ok - please seed the data to ensure that your api endpoint is working end-to-end

**Review by fbatroni — APPROVED** [review]
Approved. Overall changes look good. I fixed a few merge conflicts but this was not any of your fault. There are a lot of changes happening at once.

**Pull request merged**